File: framework/src/gmsec5/SubscriptionInfo.cpp
/*
 * Copyright 2007-2022 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/** @file SubscriptionInfo.cpp
 *
 *  @brief This file contains the opaque wrapper for subscription info used
 *  when setting up and tearing down subscriptions to the middleware.
 */

#include <gmsec5/SubscriptionInfo.h>

#include <gmsec5/internal/InternalSubscriptionInfo.h>

#include <gmsec5/Callback.h>

using namespace gmsec::api5;
using namespace gmsec::api5::internal;



SubscriptionInfo::SubscriptionInfo(InternalSubscriptionInfo* internal)
	: m_internal(internal)
{
}


SubscriptionInfo::~SubscriptionInfo()
{
}


const char* SubscriptionInfo::getSubjectPattern() const
{
	return m_internal->getSubjectPattern();
}


InternalSubscriptionInfo* SubscriptionInfo::getInternal() const
{
	return m_internal.get();
}

File: java/gov/nasa/gsfc/gmsec/api5/SubscriptionInfo.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file SubscriptionInfo.java
 */

package gov.nasa.gsfc.gmsec.api5;

import gov.nasa.gsfc.gmsec.api5.jni.JNISubscriptionInfo;


/**
 * A class that contains the information the user has supplied when
 * setting up a subscription.
 *
 * @see Connection#subscribe(String)
 * @see Connection#unsubscribe(SubscriptionInfo)
 */
public class SubscriptionInfo
{
	private JNISubscriptionInfo m_jniSubInfo;


	// Prevent users from instantiating their own objects.
	private SubscriptionInfo()
	{
	}


	//! @cond
	/**
	 * @hidden
	 * This constructor for internal GMSEC API use only.
	 * @param jSubInfo A JNISubscriptionInfo object.
	 */
	public SubscriptionInfo(JNISubscriptionInfo jSubInfo)
	{
		m_jniSubInfo = jSubInfo;
	}


	/**
	 * @hidden
	 * This method for internal GMSEC API use only.
	 * @param obj A SubscriptionInfo object.
	 * @return A JNISubscriptionInfo object.
	 */
	public static JNISubscriptionInfo getInternal(SubscriptionInfo obj)
	{
		return (obj == null ? null : obj.m_jniSubInfo);
	}
	//! @endcond


	/**
	 * Returns the subject pattern string used when setting up the subscription.
	 *
	 * @return A subject pattern string.
	 */
	public String getSubjectPattern()
	{
		return m_jniSubInfo.getSubjectPattern();
	}
}
